instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/AttributesExtractorCustomizer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import java.util.List;
+
+/**
+ * A service provider interface (SPI) for providing {@link AttributesExtractorCustomizer} instances
+ * that are conditionally applied based on the instrumentation name.
+ *
+ * <p>This allows external modules or plugins to contribute custom attribute extraction logic for
+ * specific instrumented libraries, without modifying core instrumentation code.
+ */
+public interface AttributesExtractorCustomizer {
```

Comment:
Getting this merged immediately as stable api is a tall order, consider starting with experimental incubating api first.

Reply:
If I move the class to incubating api and use it in `InstrumenterBuilder`, I need to add `api("io.opentelemetry:instrumentation-api-incubator")` in instrumentation-api. It seems cause circular dependency problem. For the problem, do you have any insight?

Reply:
If it is easier for you you could start by placing your api in an `internal` package in the `instrumentation-api` module. We can later help you move it to `instrumentation-api-incubator`. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/Experimental.java for ideas.